test/FunctionalTests/ServerStreamingMethodTests.cs

```diff
@@ -114,12 +117,55 @@ public async Task WriteResponseHeadersAsyncCore_FlushesHeadersToClient()
                 Assert.AreEqual($"How are you World? {i}", greeting.Message);
             }
 
-            var goodbyeTask = MessageHelpers.AssertReadMessageStreamAsync<HelloReply>(responseStream);
+            var goodbyeTask = MessageHelpers.AssertReadStreamMessageAsync<HelloReply>(pipeReader);
             Assert.False(goodbyeTask.IsCompleted);
             Assert.AreEqual("Goodbye World!", (await goodbyeTask.DefaultTimeout()).Message);
 
-            var finishedTask = MessageHelpers.AssertReadMessageStreamAsync<HelloReply>(responseStream);
+            var finishedTask = MessageHelpers.AssertReadStreamMessageAsync<HelloReply>(pipeReader);
             Assert.IsNull(await finishedTask.DefaultTimeout());
         }
+
+        [Test]
+        public async Task Buffering_SuccessResponsesStreamed()
```

Comment:
Is this a little redundant? ServerStreaming and DuplexStreaming mechanisms are both using a `IServerStreamWriter`

Reply:
You say that but I found a bug by adding having both tests. One of the call handlers wasn't flushing correctly at the end.